### PR TITLE
Basic examples fixes

### PR DESCRIPTION
**Pull request by ddrevicky**
## What does this PR do?

- Fixes broken links to basic_examples models in [pl_examples/README.md](https://github.com/PyTorchLightning/pytorch-lightning/tree/master/pl_examples)
- Fixes wrong paths to models in basic_examples [README](https://github.com/PyTorchLightning/pytorch-lightning/tree/master/pl_examples/basic_examples)  (e.g., mnist.py does not exist)
- Adds missing validation and test steps to `basic_examples/autoencoder.py`
- Uses `DATASET_PATHS` variable for storing downloaded datasets for examples in files where it was not used

### Random question
When defining a validation/test step could the following code be used? Or does calling training_step undesirably modify the model weights + other internal stuff?

```
# Same for test_step
def validation_step(self, batch, batch_idx):
    loss = self.training_step(batch, batch_idx)
    self.log('valid_loss', loss, on_step=True)
```

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
@ddrevicky we have swapped branches regarding upcoming feat 1.2, mind rebase on actual `master`?

**Comment by carmocca** [comment]
> When defining a validation/test step could the following code be used? Or does calling training_step undesirably modify the model weights + other internal stuff?

Yes. That can be done with no issues

**Review by carmocca — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

File: pl_examples/basic_examples/autoencoder.py [hunk 1]
```diff
@@ -64,6 +64,22 @@ def training_step(self, batch, batch_idx):
         loss = F.mse_loss(x_hat, x)
         return loss
 
+    def validation_step(self, batch, batch_idx):
+        x, y = batch
+        x = x.view(x.size(0), -1)
+        z = self.encoder(x)
+        x_hat = self.decoder(z)
+        loss = F.mse_loss(x_hat, x)
+        self.log('valid_loss', loss, on_step=True)
```
Comment (tchaton) [inline pl_examples/basic_examples/autoencoder.py:73]: There is usually no need to log on step for validation :)

**Review by tchaton — APPROVED** [review]
LGTM !
